TruckMuncher/SearchDelegate.swift

```diff
@@ -0,0 +1,103 @@
+    func searchCancelled()
+}
+
+class SearchDelegate<T: SearchCompletionProtocol where T: UIViewController>: NSObject, UISearchBarDelegate {
+    var items = [RLMObject]()
+    let searchManager = SearchManager()
+    let limit = 20
+    var offset = 0
+    var previousSearchTerm = ""
+    let completionDelegate: T
+    let searchBar: UISearchBar
+    var titleView: UIView?
+    var leftButton: UIBarButtonItem?
+    var rightButton: UIBarButtonItem?
+    
+    init(completionDelegate: T) {
+        self.completionDelegate = completionDelegate
+        searchBar = UISearchBar()
+        searchBar.showsCancelButton = true
+        searchBar.placeholder = "What are you hungry for?"
+    }
+    
+    func showSearchBar() {
+        searchBar.becomeFirstResponder()
+        titleView = self.completionDelegate.navigationItem.titleView
+        leftButton = self.completionDelegate.navigationItem.leftBarButtonItem
+        rightButton = self.completionDelegate.navigationItem.rightBarButtonItem
+        self.completionDelegate.navigationItem.titleView = searchBar
+        self.completionDelegate.navigationItem.leftBarButtonItem = nil
+        self.completionDelegate.navigationItem.rightBarButtonItem = nil
+    }
+    
+    func searchBarSearchButtonClicked(searchBar: UISearchBar) {
+        self.searchBar.resignFirstResponder()
+        let newTerm = searchBar.text
+        if newTerm == "" {
+            return
+        } else if newTerm != previousSearchTerm {
+            previousSearchTerm = newTerm
+            offset = 0
+        }
+        searchManager.simpleSearch(query: newTerm, withLimit: limit, andOffset: offset, success: { (response) -> () in
+            self.offset += self.limit
+            let results = response as [RSearch]
+            let menus = results.map { $0.menu }
+            var truckIds = NSMutableSet()
+            let trucks = results.map { $0.truck }.filter {
+                if truckIds.containsObject($0.id) {
+                    return false
+                }
+                truckIds.addObject($0.id)
+                return true
+            }
+            // we don't get back location info in search, so we have to preserve what we have in our cache
+            for truck in trucks {
+                if let rtruck = RTruck.objectsWhere("id = %@", truck.id).firstObject() as? RTruck {
+                    truck.latitude = rtruck.latitude
+                    truck.longitude = rtruck.longitude
+                }
+            }
+            let realm = RLMRealm.defaultRealm()
+            realm.beginWriteTransaction()
+            for menu in menus {
+                RMenu.createOrUpdateInRealm(realm, withObject: menu as RMenu)
+            }
+            for truck in trucks {
+                RTruck.createOrUpdateInRealm(realm, withObject: truck as RTruck)
+            }
+            realm.commitWriteTransaction()
+            // TODO the blurbs need to be combined and probably used at some point
```

Comment:
Well what's showing right now are the truck keywords. The only reason they are showing after you search is because I `didn't` pull them out of realm and they are coming straight from the API. But i figured until I get the realm issue fixed I'm not going to bother persisting arrays of things that just disappear anyways.